File: schemaorggo/postaladdress.go
package schemaorggo

import "encoding/json"

// PostalAddress see : https://schema.org/PostalAddress
type PostalAddress struct {
	ContactPoint

	typeContext

	// AddressCountry see : https://schema.org/addressCountry
	// The country. For example, USA. You can also provide the two-letter ISO 3166-1 alpha-2 country code (see: https://schema.orghttp://en.wikipedia.org/wiki/ISO_3166-1).
	// types : Country Text
	AddressCountry []interface{} `json:"addressCountry,omitempty"`

	// AddressLocality see : https://schema.org/addressLocality
	// The locality. For example, Mountain View.
	// types : Text
	AddressLocality []string `json:"addressLocality,omitempty"`

	// AddressRegion see : https://schema.org/addressRegion
	// The region. For example, CA.
	// types : Text
	AddressRegion []string `json:"addressRegion,omitempty"`

	// PostOfficeBoxNumber see : https://schema.org/postOfficeBoxNumber
	// The post office box number for PO box addresses.
	// types : Text
	PostOfficeBoxNumber []string `json:"postOfficeBoxNumber,omitempty"`

	// PostalCode see : https://schema.org/postalCode
	// The postal code. For example, 94043.
	// types : Text
	PostalCode []string `json:"postalCode,omitempty"`

	// StreetAddress see : https://schema.org/streetAddress
	// The street address. For example, 1600 Amphitheatre Pkwy.
	// types : Text
	StreetAddress []string `json:"streetAddress,omitempty"`
}

func (v PostalAddress) IntoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.ContactPoint.IntoMap(intop)

	into := *intop

	{
		var value interface{} = v.AddressCountry
		if len(v.AddressCountry) == 1 {
			value = v.AddressCountry[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["addressCountry"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.AddressLocality
		if len(v.AddressLocality) == 1 {
			value = v.AddressLocality[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["addressLocality"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.AddressRegion
		if len(v.AddressRegion) == 1 {
			value = v.AddressRegion[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["addressRegion"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.PostOfficeBoxNumber
		if len(v.PostOfficeBoxNumber) == 1 {
			value = v.PostOfficeBoxNumber[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["postOfficeBoxNumber"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.PostalCode
		if len(v.PostalCode) == 1 {
			value = v.PostalCode[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["postalCode"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.StreetAddress
		if len(v.StreetAddress) == 1 {
			value = v.StreetAddress[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["streetAddress"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v PostalAddress) AsMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.IntoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "PostalAddress"

	return data, nil
}

func (v PostalAddress) MarshalJSON() ([]byte, error) {
	data, err := v.AsMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: releases/v3.0.0/schemaorgphp/classes/postal_address.php
<?php
namespace SchemaOrg;

// PostalAddress see : https://schema.org/PostalAddress
class PostalAddress implements \JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type    = 'PostalAddress';

	/**
	 * With properties from ContactPoint see : https://schema.org/ContactPoint
	 */

	/**
	 * With properties from StructuredValue see : https://schema.org/StructuredValue
	 */

	/**
	 * With properties from Intangible see : https://schema.org/Intangible
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */

	/**
	 * With properties from Intangible see : https://schema.org/Intangible
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */


	/**
	 * An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	 * see : https://schema.org/additionalType
	 *
	 * @var string | string[]
	 */
	public $additional_type;

	/**
	 * The country. For example, USA. You can also provide the two-letter &lt;a href=&#39;http://en.wikipedia.org/wiki/ISO_3166-1&#39;&gt;ISO 3166-1 alpha-2 country code&lt;/a&gt;.
	 * see : https://schema.org/addressCountry
	 *
	 * @var string | string[] | \Country | \Country[]
	 */
	public $address_country;

	/**
	 * The locality. For example, Mountain View.
	 * see : https://schema.org/addressLocality
	 *
	 * @var string | string[]
	 */
	public $address_locality;

	/**
	 * The region. For example, CA.
	 * see : https://schema.org/addressRegion
	 *
	 * @var string | string[]
	 */
	public $address_region;

	/**
	 * An alias for the item.
	 * see : https://schema.org/alternateName
	 *
	 * @var string | string[]
	 */
	public $alternate_name;

	/**
	 * The geographic area where a service or offered item is provided.
	 * see : https://schema.org/areaServed
	 *
	 * @var \Place | \Place[] | \AdministrativeArea | \AdministrativeArea[] | \GeoShape | \GeoShape[] | string | string[]
	 */
	public $area_served;

	/**
	 * A language someone may use with the item. Please use one of the language codes from the &lt;a href=&#39;http://tools.ietf.org/html/bcp47&#39;&gt;IETF BCP 47 standard&lt;/a&gt;. See also [[inLanguage]].
	 * see : https://schema.org/availableLanguage
	 *
	 * @var \Language | \Language[] | string | string[]
	 */
	public $available_language;

	/**
	 * An option available on this contact point (e.g. a toll-free number or support for hearing-impaired callers).
	 * see : https://schema.org/contactOption
	 *
	 * @var \ContactPointOption | \ContactPointOption[]
	 */
	public $contact_option;

	/**
	 * A person or organization can have different contact points, for different purposes. For example, a sales contact point, a PR contact point and so on. This property is used to specify the kind of contact point.
	 * see : https://schema.org/contactType
	 *
	 * @var string | string[]
	 */
	public $contact_type;

	/**
	 * A description of the item.
	 * see : https://schema.org/description
	 *
	 * @var string | string[]
	 */
	public $description;

	/**
	 * A sub property of description. A short description of the item used to disambiguate from other, similar items. Information from other properties (in particular, name) may be necessary for the description to be useful for disambiguation.
	 * see : https://schema.org/disambiguatingDescription
	 *
	 * @var string | string[]
	 */
	public $disambiguating_description;

	/**
	 * Email address.
	 * see : https://schema.org/email
	 *
	 * @var string | string[]
	 */
	public $email;

	/**
	 * The fax number.
	 * see : https://schema.org/faxNumber
	 *
	 * @var string | string[]
	 */
	public $fax_number;

	/**
	 * The hours during which this service or contact is available.
	 * see : https://schema.org/hoursAvailable
	 *
	 * @var \OpeningHoursSpecification | \OpeningHoursSpecification[]
	 */
	public $hours_available;

	/**
	 * An image of the item. This can be a &lt;a href=&quot;http://schema.org/URL&quot;&gt;URL&lt;/a&gt; or a fully described &lt;a href=&quot;http://schema.org/ImageObject&quot;&gt;ImageObject&lt;/a&gt;.
	 * see : https://schema.org/image
	 *
	 * @var string | string[] | \ImageObject | \ImageObject[]
	 */
	public $image;

	/**
	 * Indicates a page (or other CreativeWork) for which this thing is the main entity being described. See &lt;a href=&quot;/docs/datamodel.html#mainEntityBackground&quot;&gt;background notes&lt;/a&gt; for details.
	 * see : https://schema.org/mainEntityOfPage
	 *
	 * @var \CreativeWork | \CreativeWork[] | string | string[]
	 */
	public $main_entity_of_page;

	/**
	 * The name of the item.
	 * see : https://schema.org/name
	 *
	 * @var string | string[]
	 */
	public $name;

	/**
	 * The post office box number for PO box addresses.
	 * see : https://schema.org/postOfficeBoxNumber
	 *
	 * @var string | string[]
	 */
	public $post_office_box_number;

	/**
	 * The postal code. For example, 94043.
	 * see : https://schema.org/postalCode
	 *
	 * @var string | string[]
	 */
	public $postal_code;

	/**
	 * Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	 * see : https://schema.org/potentialAction
	 *
	 * @var \Action | \Action[]
	 */
	public $potential_action;

	/**
	 * The product or service this support contact point is related to (such as product support for a particular product line). This can be a specific product or product line (e.g. &quot;iPhone&quot;) or a general category of products or services (e.g. &quot;smartphones&quot;).
	 * see : https://schema.org/productSupported
	 *
	 * @var \Product | \Product[] | string | string[]
	 */
	public $product_supported;

	/**
	 * URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Freebase page, or official website.
	 * see : https://schema.org/sameAs
	 *
	 * @var string | string[]
	 */
	public $same_as;

	/**
	 * The geographic area where the service is provided.
	 * see : https://schema.org/serviceArea
	 *
	 * @var \Place | \Place[] | \AdministrativeArea | \AdministrativeArea[] | \GeoShape | \GeoShape[]
	 */
	public $service_area;

	/**
	 * The street address. For example, 1600 Amphitheatre Pkwy.
	 * see : https://schema.org/streetAddress
	 *
	 * @var string | string[]
	 */
	public $street_address;

	/**
	 * The telephone number.
	 * see : https://schema.org/telephone
	 *
	 * @var string | string[]
	 */
	public $telephone;

	/**
	 * URL of the item.
	 * see : https://schema.org/url
	 *
	 * @var string | string[]
	 */
	public $url;

	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type'    => 'PostalAddress',
		);

		$serialized = \SchemaOrg\json_serialize( $this->additional_type );
		if ( ! empty( $serialized ) ) {
			$out['additionalType'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->address_country );
		if ( ! empty( $serialized ) ) {
			$out['addressCountry'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->address_locality );
		if ( ! empty( $serialized ) ) {
			$out['addressLocality'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->address_region );
		if ( ! empty( $serialized ) ) {
			$out['addressRegion'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->alternate_name );
		if ( ! empty( $serialized ) ) {
			$out['alternateName'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->area_served );
		if ( ! empty( $serialized ) ) {
			$out['areaServed'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->available_language );
		if ( ! empty( $serialized ) ) {
			$out['availableLanguage'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->contact_option );
		if ( ! empty( $serialized ) ) {
			$out['contactOption'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->contact_type );
		if ( ! empty( $serialized ) ) {
			$out['contactType'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->description );
		if ( ! empty( $serialized ) ) {
			$out['description'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->disambiguating_description );
		if ( ! empty( $serialized ) ) {
			$out['disambiguatingDescription'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->email );
		if ( ! empty( $serialized ) ) {
			$out['email'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->fax_number );
		if ( ! empty( $serialized ) ) {
			$out['faxNumber'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->hours_available );
		if ( ! empty( $serialized ) ) {
			$out['hoursAvailable'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->image );
		if ( ! empty( $serialized ) ) {
			$out['image'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->main_entity_of_page );
		if ( ! empty( $serialized ) ) {
			$out['mainEntityOfPage'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->name );
		if ( ! empty( $serialized ) ) {
			$out['name'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->post_office_box_number );
		if ( ! empty( $serialized ) ) {
			$out['postOfficeBoxNumber'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->postal_code );
		if ( ! empty( $serialized ) ) {
			$out['postalCode'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->potential_action );
		if ( ! empty( $serialized ) ) {
			$out['potentialAction'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->product_supported );
		if ( ! empty( $serialized ) ) {
			$out['productSupported'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->same_as );
		if ( ! empty( $serialized ) ) {
			$out['sameAs'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->service_area );
		if ( ! empty( $serialized ) ) {
			$out['serviceArea'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->street_address );
		if ( ! empty( $serialized ) ) {
			$out['streetAddress'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->telephone );
		if ( ! empty( $serialized ) ) {
			$out['telephone'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->url );
		if ( ! empty( $serialized ) ) {
			$out['url'] = $serialized;
		}

		return $out;
	}
}
